### ISSUE TITLE
Add native functions to module info

### ISSUE THREAD

**Issue by garazdawi** [erlang/otp#9072]
**Is your feature request related to a problem? Please describe.**
We need a way to get information about which functions in a module is backed by native code, both nifs or bifs.

**Describe the solution you'd like**
the funcion `lists:module_info(native_functions)` should return a list of functions that are implemented using native code. It should return both exported and private functions.

**Additional context**
#9010

**Comment by josevalim** [comment]
@garazdawi is there a reason to not use the current `lists:module_info(nifs)`? From an external point of view, NIFs vs BIFs are mostly an implementation detail, no? :)

**Comment by josevalim** [comment]
Alternatively, what about exposing `-nifs` under `lists:module_info(attributes)`? That would provide a quick way for fetching all NIFs, without having to traverse AST (or without having to load the module, as we could read it from the chunk).

**Comment by garazdawi** [comment]
> is there a reason to not use the current lists:module_info(nifs)? 

Nope, we can definitely use that for also BIFs. I just did not remember that it existed :)

> Alternatively, what about exposing -nifs under lists:module_info(attributes)? That would provide a quick way for fetching all NIFs, without having to traverse AST (or without having to load the module, as we could read it from the chunk).

The `-nifs` attribute only shows which function could potentially be NIFs, and not which must be NIFs, so that cannot be used.
